react-client/src/Components/App.js

```diff
@@ -62,12 +60,18 @@ function App() {
           };
         });
 
-        var eltarget = e.target;
+        var eltarget = e.target;        
 
         var randomCode = uuidv4();
      
-        e.target.setAttribute("data-id" , randomCode);                  
+        e.target.setAttribute("data-id" , randomCode);
        
+        var attributekey = new Array();
+        var attributevalue = new Array(); 
+        Object.entries(e.target.dataset).forEach(([key, value]) => attributekey.push(key));
+        Object.entries(e.target.dataset).forEach(([key, value]) => attributevalue.push(value));
```

Comment:
Redundant data attribute mapping. Only use one instead.

Use this to solve the problem: `const dataAttributes = Object.entries(e.target.dataset).reduce((acc, [key, value]) => acc.concat([{ key, value }]), []);`

Take note that you'll still need to merge both `attributekey` and `attributevalue` into one object. Name it `dataAttributes` instead and put it in domInfo state.